woocommerce/woocommerce
# Add API tests for creating, retrieving, updating, and deleting products

### All Submissions:


### Changes proposed in this Pull Request:



Closes the following:
- https://github.com/woocommerce/woocommerce/issues/30662
- https://github.com/woocommerce/woocommerce/issues/30664
- https://github.com/woocommerce/woocommerce/issues/30665
- https://github.com/woocommerce/woocommerce/issues/30666
- https://github.com/woocommerce/woocommerce/issues/30668
- https://github.com/woocommerce/woocommerce/issues/30669

It also renames the `products.test.js` file to `product-list.test.js` to distinguish it from the tests in this PR, which mostly focuses on single products, and a few products in the batch update tests.

### How to test the changes in this Pull Request:

1. Clone this branch, and run the setup scripts for the e2e environment.
2. Run the API tests for Products: `npm test -- --group=products`.
3. All the tests in `products-crud.test.js` and `product-list.test.js` should pass:
```
 PASS  tests/products/products-crud.test.js
  Products API tests: CRUD
    ✓ can add a simple product (168ms)
    ✓ can add a virtual product (257ms)
    ✓ can add a variable product (277ms)
    ✓ can view a single product (88ms)
    ✓ can update a single product (116ms)
    ✓ can delete a product (207ms)
    Batch update products
      ✓ can batch create products (177ms)
      ✓ can batch update products (144ms)
      ✓ can batch delete products (290ms)

 PASS  tests/products/product-list.test.js (11.775s)
  Products API tests: List All Products
    List all products
      ✓ defaults (134ms)
      ✓ pagination (464ms)
      ✓ search (167ms)
      ✓ inclusion / exclusion (370ms)
      ✓ slug (177ms)
      ✓ sku (163ms)
      ✓ type (452ms)
      ✓ featured (240ms)
      ✓ categories (240ms)
      ✓ on sale (264ms)
      ✓ price (296ms)
      ✓ before / after (274ms)
      ✓ attributes (118ms)
      ✓ status (219ms)
      ✓ shipping class (99ms)
      ✓ tax class (102ms)
      ✓ stock status (99ms)
      ✓ tags (100ms)
      ✓ parent (231ms)
      orderby
        ✓ default (127ms)
        ✓ date (122ms)
        ✓ id (229ms)
        ✓ title (271ms)
        ✓ slug (283ms)
        ✓ price (275ms)
        ✓ include (215ms)
        ✓ rating (desc) (89ms)
        ✓ popularity (desc) (98ms)
        ○ skipped rating (asc)
        ○ skipped popularity (asc)

Test Suites: 2 passed, 2 of 6 total
Tests:       2 skipped, 37 passed, 39 total
```
4. Run `curl --location -u admin:password --request GET 'http://localhost:8084/wp-json/wc/v3/products'` to query the list of products. It should return an empty list. This means that all the products used by the tests were correctly cleaned up.




### Changelog entry

> Added tests for creating, retrieving, updating, and deleting products

### FOR PR REVIEWER ONLY:
